File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
       int ans = nums[0];
       int cnt = 0;
       for(auto x:nums)
       {
            if(ans == x)
                cnt++;
            else
                cnt--;
            if(cnt == 0)
            {
                cnt = 1;
                ans = x;
            }
       }
       return ans;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        
        cnt = 0
        res = -1
        curr = nums[0]

        for i in nums:
            if i == curr:
                cnt+=1
            else:
                cnt-=1

                if cnt < 0:
                    curr = i
                    cnt = 1
        return curr